### DISCUSSION TITLE
How to style light/dark mode for expo router navigation components?

### DISCUSSION THREAD

**Discussion by chhornponleu** [nativewind/nativewind#705] [Q&A]
I am aware of react-navigation options in Expo Router, but we cannot share existing color variable between (global).css and the navigation settings.

**Comment by harshit9715** [comment]
try creating a blank app using create-expo-stack cli... when prompted use nativewindUI for the configuration. their project setup has all the defauls that can show you how it works.

or you can checkout nativewind-ui documentation.

**Comment by shovel-kun** [comment]
CSS Interop works for me

See:
- https://github.com/hoarder-app/hoarder/blob/300f3c5d0b661c430ad2f6b479b151ec65f14243/apps/mobile/components/navigation/tabs.tsx
- https://github.com/hoarder-app/hoarder/blob/300f3c5d0b661c430ad2f6b479b151ec65f14243/apps/mobile/components/navigation/stack.tsx

**Reply by rachelslurs** [reply]
OMG THANK YOU! I've been hitting my head on the wall over this because of those darn nested properties!

**Reply by rachelslurs** [reply]
I feel like this ought to be somewhere more prominent because I've spent way too long trying to figure this out. Doing things like this did not work because of the nested nature of it:

```
// DOES NOT WORK
const CustomTabs = cssInterop(Tabs, {
  tabBarClassName: "screenOptions.tabBarStyle",
});
```

```
// DOES NOT WORK
const CustomTabs = remapProps(Tabs, {
  tabBarClassName: "screenOptions.tabBarStyle",
});
```

but that component you linked to is brilliant! Copying it here just in case:

```
// WORKS! 🎉
import { ViewStyle } from "react-native";
import { Tabs } from "expo-router";
import { cssInterop } from "nativewind";

function StyledTabsImpl({
  tabBarStyle,
  headerStyle,
  ...props
}: React.ComponentProps<typeof Tabs> & {
  tabBarStyle?: ViewStyle;
  headerStyle?: ViewStyle;
}) {
  props.screenOptions = {
    ...props.screenOptions,
    tabBarStyle,
    headerStyle,
  };
  return <Tabs {...props} />;
}

export const StyledTabs = cssInterop(StyledTabsImpl, {
  tabBarClassName: "tabBarStyle",
  headerClassName: "headerStyle",
  sceneContainerClassName: "sceneContainerStyle",
});
```

**Comment by danstepanov** [comment]
https://www.nativewind.dev/docs/core-concepts/dark-mode
